ResearchUI/ResearchUI/iOS/RSDTaskViewController.swift

```diff
@@ -163,55 +170,77 @@ open class RSDTaskViewController: UIViewController, RSDTaskUIController, UIPageV
     ///
     /// This method will look for a step view controller in the following order:
-    /// 1. Call the delegate method `taskViewController(:, viewControllerFor step:)` and return the view controller
-    ///    supplied by that method. This will throw an exception if the delegate does not return a view controller
-    ///    that conforms to the `RSDStepController` protocol and inherits from `UIViewController`.
-    /// 2. If the given step implements the `RSDStepViewControllerVendor` protocol and returns a non-nil instance of
-    ///    a view controller, then that will be returned.
-    /// 3. If the given step implements the `RSDThemedUIStep` protocol and returns a `viewTheme`, then call
+    /// 1. Call the delegate method `taskViewController(:, viewControllerForStep stepModel:)` and return the
+    ///    view controller supplied by that method. This will throw an exception if the delegate does not
+    ///    return a view controller that conforms to the `RSDStepController` protocol and inherits from
+    ///    `UIViewController`.
+    /// 2. If the given step implements the `RSDThemedUIStep` protocol and returns a `viewTheme`, then call
     ///    `instantiateViewController(with viewTheme:)` to instantiate a view controller for this step.
-    /// 4. If none of the functions listed above returns a view controller then return the view controller instantiated
-    ///    by calling `vendDefaultViewController(for step:)`.
+    /// 3. If the given step implements the `RSDStepViewControllerVendor` protocol and returns a non-nil
+    ///    instance of a view controller, then that will be returned.
+    /// 4. If none of the functions listed above returns a view controller then return the view controller
+    ///    instantiated by calling `vendDefaultViewController(for step:, with parent:)`.
     ///
     /// - parameter step: The step to display.
     /// - returns: The view controller to use when displaying a given step.
-    open func viewController(for step: RSDStep) -> (UIViewController & RSDStepController) {
+    open func stepController(for step: RSDStep, with parent: RSDPathComponent?) -> RSDStepController? {
+        guard let controller = _stepController(for: step, with: parent) else { return nil }
+        if controller.stepViewModel == nil {
+            controller.stepViewModel = RSDStepViewModel(step: step, parent: parent)
+        }
+        return controller
+    }
+    
+    private func _stepController(for step: RSDStep, with parent: RSDPathComponent?) -> RSDStepController? {
         // Exit early if the delegate, step or storyboard returns a view controller
+        if let vc = delegate?.taskViewController?(self, viewControllerForStep: RSDStepViewModel(step: step, parent: parent)) {
+            return (vc as! RSDStepController)
+        }
+        // TODO: syoung 09/07/2018 remove the deprecated method once refactor is completed.
```

Comment:
Is this still todo?

Reply:
I was thinking I would leave it in for a cycle.